incubator/ethereum/values.yaml

```diff
@@ -0,0 +1,37 @@
+# Default values for ethereum.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+imagePullPolicy: IfNotPresent
+
+bootnode:
+  image:
+    repository: ethereum/client-go
+    tag: alltools-v1.7.3
+
+ethstats:
+  image:
+    repository: ethereumex/eth-stats-dashboard
+    tag: latest
+  webSocketSecret: my-secret-for-connecting-to-ethstats
+  service:
+    type: LoadBalancer
+
+geth:
+  image:
+    repository: ethereum/client-go
+    tag: v1.7.3
+  tx:
+    # transaction nodes
+    replicaCount: 2
+    service:
+      type: ClusterIP
+  miner:
+    # miner nodes
+    replicaCount: 3
+    account:
+      secret: my-secret-account-password
+  genesis:
+    networkId: 98052
+    accountsToFund:
+      - da1e55471ded03adadee63ac658b671277a6c7d1
```

Comment:
It appears there's a default account to start funding. From our tests and someone who installs this without setting up this own account people could end up mining (using their resources) for someone else.

This should use a pattern of not supplying a default account to fund. If that is required the parts that require it should be skipped and the NOTES.txt file should supply detail on how to handle this.

Reply:
Hey @mattfarina. Thanks for the comment.


> From our tests and someone who installs this without setting up this own account people could end up mining (using their resources) for someone else.

I want to clarify that this chart will deploy a *private* network so it is not connected to mainnet. What happens inside this Helm chart all stays within the k8s cluster. It is a *separate* Ethereum network.

Not sure how familiar you are with Ethereum, but to give you with a bit more detail:

* Mining
When the geth-miners initialize, they run a `create-account` command (https://github.com/jpoon/charts/blob/master/incubator/ethereum/templates/geth-miner.deployment.yaml#L64). This command creates a new geth account and outputs a public/private key. It is this account where the mined Ether will be deposited to, and given the private key, you have access said mined Ether. However, as this is a private network (ie. you have control over everything), I would recommend that you leave these accounts alone and instead, configure a set number of accounts with a sufficent amount of Ether for usage (ie. to pay for contract mining). You'll see here I initialize these accounts with a crazy amount of Ether (https://github.com/jpoon/charts/blob/master/incubator/ethereum/templates/geth.configmap.yaml#L26).

* accountsToFund
 This configuration accepts an array of public keys and will initialize these accounts with enough Ether to last a lifetime and be sufficient to pay for things like mining of contract and the like. The intention is for a user to supply these values as, once the Ethereum network is initialized, you can't go back. It's the nature of the blockchain that you can't reconfigure these types of settings. I intentionally left this configuration in the `values.yaml` as it makes it easier for somebody to notice and realize that it's something they should change so I'm glad you noticed :)

Reply:
It's the seed value, so, it doesn't seem very dangerous - but I'd echo Matt's sentiment here. It seems awfully easy to forget to change this.

Also, one could start with a genesis.json that has multiple allocs defined already correct? Is there benefit to moving this out of that config into a separate parameter?

Reply:
> one could start with a genesis.json that has multiple allocs defined already correct?

The configuration accepts an array, so supplying multiple allocs is fine. Not sure what difference supply one vs multiple default accounts would make? 

In my latest commit, I've also added the private key for the default account to fund, and included instructions on how to use: https://github.com/jpoon/charts/tree/master/incubator/ethereum/key. I was debating making this a k8s secret, but I feel like this would obscure how to use it even more.

>  Is there benefit to moving this out of that config into a separate parameter?

Not sure I understand your comment, it is already in a separate configuration/parameter? In the README, there's a [table of all the configurations](https://github.com/jpoon/charts/blob/master/incubator/ethereum/README.md#configuration):

```
helm install ethereum --set geth.genesis.accountsToFund=ACCOUNT1, ACCOUNT2, blah
```